llvm/lib/Target/X86/X86ISelLowering.cpp

```diff
@@ -24790,8 +24790,8 @@ static SDValue LowerVACOPY(SDValue Op, const X86Subtarget &Subtarget,
   return DAG.getMemcpy(
       Chain, DL, DstPtr, SrcPtr,
       DAG.getIntPtrConstant(Subtarget.isTarget64BitLP64() ? 24 : 16, DL),
-      Align(Subtarget.isTarget64BitLP64() ? 8 : 4), /*isVolatile*/ false, false,
-      false, MachinePointerInfo(DstSV), MachinePointerInfo(SrcSV));
+      Align(Subtarget.isTarget64BitLP64() ? 8 : 4), /*Vol=*/{false, false},
```

Comment:
For understandability, should we use a MemTransferVolatility builder helper instead of embedding implicit values like this?

Reply:
good idea, something like `MemTransferVolatility().setDst(true/false).setSrc(true/false)` ? Do you have preference for names?

Reply:
No preference for names - do we need to handle runtime bool values or could a few basic hardcoded combos be enough MemTransferVolatility::neverVolatile() / MemTransferVolatility::readVolatile() ?

Reply:
Just pushed a patch that uses `MemTransferVolatility` default ctor for the non-volatile case, and allows `.Dst(bool)`, `.Src(bool)` and/or `.All(bool)` to explicitly specify.  One does need the ability to provide runtime values there.